core/src/main/java/io/questdb/griffin/engine/functions/date/TodayWithTimezoneFunctionFactory.java

```diff
@@ -0,0 +1,178 @@
+
+        public Func(Function tzFunc) {
+            this.tzFunc = tzFunc;
+        }
+
+        @Override
+        public Function getArg() {
+            return tzFunc;
+        }
+
+        @Override
+        public @NotNull Interval getInterval(Record rec) {
+            long nowWithTz = now;
+            final CharSequence tz = tzFunc.getStrA(rec);
+            if (tz != null) {
+                try {
+                    nowWithTz = Timestamps.toTimezone(nowWithTz, TimestampFormatUtils.EN_LOCALE, tz);
+                } catch (NumericException e) {
+                    return Interval.NULL;
+                }
+            }
+            final long todayStart = Timestamps.floorDD(nowWithTz);
+            final long todayEnd = Timestamps.floorDD(Timestamps.addDays(nowWithTz, 1)) - 1;
+            return interval.of(todayStart, todayEnd);
+        }
+
+        @Override
+        public void init(SymbolTableSource symbolTableSource, SqlExecutionContext executionContext) throws SqlException {
+            UnaryFunction.super.init(symbolTableSource, executionContext);
+            now = executionContext.getNow();
+        }
+
+        @Override
+        public boolean isConstant() {
+            return false;
+        }
+
+        @Override
+        public boolean isReadThreadSafe() {
+            return UnaryFunction.super.isReadThreadSafe();
+        }
+
+        @Override
+        public boolean isRuntimeConstant() {
+            return UnaryFunction.super.isRuntimeConstant();
+        }
+
+        @Override
+        public void toPlan(PlanSink sink) {
+            sink.val(SIGNATURE);
+        }
+    }
+
+    private static class RuntimeConstFunc extends IntervalFunction implements UnaryFunction {
+        private final Interval interval = new Interval();
+        private final Function tzFunc;
+
+        public RuntimeConstFunc(Function tzFunc) {
+            this.tzFunc = tzFunc;
+        }
+
+        @Override
+        public Function getArg() {
+            return tzFunc;
+        }
+
+        @Override
+        public @NotNull Interval getInterval(Record rec) {
+            return interval;
+        }
+
+        @Override
+        public void init(SymbolTableSource symbolTableSource, SqlExecutionContext executionContext) throws SqlException {
+            UnaryFunction.super.init(symbolTableSource, executionContext);
+            long now = executionContext.getNow();
+            final CharSequence tz = tzFunc.getStrA(null);
+            if (tz != null) {
+                try {
+                    now = Timestamps.toTimezone(now, TimestampFormatUtils.EN_LOCALE, tz);
+                } catch (NumericException e) {
+                    interval.of(Interval.NULL.getLo(), Interval.NULL.getHi());
+                    return;
+                }
+            }
+            long todayStart = Timestamps.floorDD(now);
+            long todayEnd = Timestamps.floorDD(Timestamps.addDays(now, 1)) - 1;
+            interval.of(todayStart, todayEnd);
```

Comment:
This is suspiciously incorrect. If I want to get today('CEST') I want an interval starting from previous date 23:00 till today's date 23:00 in winter and from 22:00 to 22:00 in summer. I think this implementation is incorrect

Reply:
Hmm, let me double check these functions.

Reply:
Indeed the functions were incorrect. Fixed them and improved tests in 7353ee7363532f6a231e919aa6a02b955d97cad7

Reply:
Please test cases around the time shift in the timezone. Some days are 23 UTC hours and some are 25 UTC hours, I don't think this implementation handles these corner cases.

Reply:
Good point. I'll add the tests and tweak the impl.

Reply:
The impl was fixed in 7057561cb3f7074d2594e93e1c2658bd69201632
I'll add tests around time shifts in the timezone.

Reply:
Tests around DST switch were added in 72ba40992bcb606a5fe307680e607501452e1f15